manualCreditor.go

```diff
@@ -80,6 +80,11 @@ func (mc *manualCreditor) Drain(ctx context.Context, l *link) error {
 	case <-drained:
 		return nil
 	case <-l.Detached:
+		// NOTE: need to be careful to not pass back a nil *Error when you're
+		// returning an `error` interface!
+		if l.detachError == nil {
```

Comment:
Can we change the type of `l.detachError` to `error`, or will that get in the way of other stuff?

Reply:
Hm...let me check.